packages/collector/src/announceCycle/agentready.js

```diff
@@ -71,29 +81,33 @@ module.exports = exports = {
  */
 function enter(_ctx) {
   ctx = _ctx;
-  uncaught.activate();
-  metrics.activate();
-  initializedTooLate.check();
-  transmissionCycle.activate(
-    metrics,
-    agentConnection,
-    /**
-     * @param {Array.<import('../agent/requestHandler').AnnounceRequest>} requests
-     */
-    function onSuccess(requests) {
-      requestHandler.handleRequests(requests);
-    },
-    function onError() {
-      ctx.transitionTo('unannounced');
-    }
-  );
+
+  if (isMainThread) {
+    uncaught.activate();
+    metrics.activate();
+    initializedTooLate.check();
```

Comment:
I was wondering why we don't want to send the initialized monitoring event on a worker thread.
My impression was that a worker thread can require instana in a worker thread, which means the feature might be helpful there too.

Reply:
Good catch! That's absolutely correct, I'll move that one out of the "if (isMainThread)".